test/e2e/clusterctl_upgrade_test.go

```diff
@@ -110,10 +110,10 @@ var _ = Describe("When testing clusterctl upgrades (v0.3=>v1.5=>current)", func(
 	})
 })
 
-var _ = Describe("When testing clusterctl upgrades (v0.4=>v1.6=>current)", func() {
-	// We are testing v0.4=>v1.6=>current to ensure that old entries with v1alpha4 in managed files do not cause issues
+var _ = Describe("When testing clusterctl upgrades (v0.4=>v1.7=>current)", func() {
+	// We are testing v0.4=>v1.7=>current to ensure that old entries with v1alpha4 in managed files do not cause issues
 	// as described in https://github.com/kubernetes-sigs/cluster-api/issues/10051.
-	// NOTE: The combination of v0.4=>v1.6=>current allows us to verify this without being forced to upgrade
+	// NOTE: The combination of v0.4=>v1.7=>current allows us to verify this without being forced to upgrade
 	// the management cluster in the middle of the test as all 3 versions are ~ compatible with the same mgmt and workload Kubernetes versions.
 	// Additionally, clusterctl v1.6 still allows the upgrade of management clusters from v1alpha4 (v1.7 doesn't).
```

Comment:
```suggestion
```

I think this is simply not true, otherwise we would not be able to change this test here as is.

Reply:
cc @sbueringer wdyt?

Reply:
> cc @sbueringer wdyt?

I'll update this later based on your & @sbueringer comments!

Reply:
Not sure if we ran the test after this change, let's see

Reply:
Pretty sure my statement is still correct and the test will fail

Reply:
Context:
// Additionally, clusterctl v1.5 still allows the upgrade of management clusters from v1alpha3 (v1.6 doesn't).
https://github.com/kubernetes-sigs/cluster-api/blob/release-1.5/cmd/clusterctl/client/upgrade.go#L151-L159

// Additionally, clusterctl v1.6 still allows the upgrade of management clusters from v1alpha4 (v1.7 doesn't).
https://github.com/kubernetes-sigs/cluster-api/blob/release-1.6/cmd/clusterctl/client/upgrade.go#L149-L155

For comparison 1.7 only supports v1beta1
https://github.com/kubernetes-sigs/cluster-api/blob/release-1.7/cmd/clusterctl/client/upgrade.go#L145-L148

If I'm correct (and the test should prove it). Let's please add the links to the release-1.5/release-1.6 to the corresponding comments

Reply:
So should we change those tests back to `(v0.3=>v1.5=>current)` and `(v0.4=>v1.6=>current)` ?

Reply:
Yes